### PR TITLE
Add sub-record example and fix idl

### PR DESCRIPTION
**Pull request by kenchris**
@beaufortfrancois done!

(Support for custom sub-record types will be added soon)

***
<a href="https://pr-preview.s3.amazonaws.com/kenchris/web-nfc/pull/334.html" title="Last updated on Sep 3, 2019, 1:32 PM UTC (e1fef8f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/web-nfc/334/92cd2b0...kenchris:e1fef8f.html" title="Last updated on Sep 3, 2019, 1:32 PM UTC (e1fef8f)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -909,6 +909,85 @@ <h4>
...
+        Instead of using property names as identifiers as in JSON,
+        seperate sub-records each with their own type, will be used.
+        Custom types need to be at least one letter long, meaning the
+        overhead per record can be as little as 4 bytes, which is
+        similar to JSON, which uses 4-6 bytes per single letter property name,
+        depending on whether it is the first property or not (given , and
+        newline separators).
+      </p>
+      <p>
+        This example shows how to read an exernal record for social posts,
```
Comment (zolkis) [inline index.html:945]: typo: external

File: index.html [hunk 2]
```diff
@@ -909,6 +909,85 @@ <h4>
 setTimeout(() => controller.abort(), 3000);
     </pre>
   </aside>
+
+  <aside title="Read a external record with sub-records" class="example">
+      <p>
```
Comment (zolkis) [inline index.html:914]: I contest the usefulness of this vs the added complexity.
If storage efficiency is of concern, an external type with binary (e.g. CBOR) would be a yet more efficient solution, without the arbitrary conventions made down this example.

Comment (kenchris) [reply index.html:914]: It is supported by NDEF and even used in standard records like smart poster.

You can of course contest using NDEF at all (and write data directly to the tag using the type specific protocols), but hey that is what we are spec'ing.

File: index.html [hunk 3]
```diff
@@ -909,6 +909,85 @@ <h4>
 setTimeout(() => controller.abort(), 3000);
     </pre>
   </aside>
+
+  <aside title="Read a external record with sub-records" class="example">
```
Comment (zolkis) [inline index.html:913]: typo: an

Comment (kenchris) [reply index.html:913]: thx

File: index.html (hunk 2 above)
Comment (zolkis) [reply index.html:914]: If I am reading correctly the NFC Forum Smart Poster spec, there is no word about "sub-record", only about records (of type Title, URI, Action, Icon, Size, Type). The term "sub-record" seems to be made up outside of NFC Forum, so it's an application level term from NFC point of view. Read my previous comment in that context: apps could as well choose a more efficient storage method inside the allowed NDEF records. If we expose an app-level mechanism on top of NDEF in the Web NFC spec that needs to go though the usual feature addition path: use case, issue, group discussion, PR.

Comment (kenchris) [reply index.html:914]: OK, it definitely embeds the records and that is also defined how to do in other NFC specs. Whether we call them sub records or embedded records, I don't care.

**Pull request merged**